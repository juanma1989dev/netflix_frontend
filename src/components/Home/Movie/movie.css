.Movie_card{
    width: 300px;
}

.card-img-top {
    height: 25rem;
}