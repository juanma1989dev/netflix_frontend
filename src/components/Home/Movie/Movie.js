import React, {Component} from 'react'
import {Link} from 'react-router-dom';

import './movie.css';

class Movie extends Component{
    render(){
        return(
            <div>
                <div className="col-md d-flex aling-items-strech mt-5">
                    <div className="card Movie_card" >
                        <img className="card-img-top" src={this.props.poster} alt="Card image cap" />
                        <div className="card-body">
                            <h5 className="card-title">{this.props.title}</h5>
                            <Link className="btn btn-primary" to={`/movie/${this.props.id}`}>Ver</Link>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default Movie;