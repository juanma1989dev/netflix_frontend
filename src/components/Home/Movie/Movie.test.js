import React from 'react';
import {shallow} from 'enzyme';
import Movie from './Movie';


describe("Movie", () => {

    const movieObject = {
        "poster": "www.google.com",
        "title" : "La pelicula de test",
        "id"    : "auayuasgsydaguagduavdghasvdhgsa"
    }

    const component = shallow(<Movie title={movieObject.title} poster={movieObject.poster} 
                                     id={movieObject.id}  />)
    
    it("Check if props passed correctly", ()=>{
        expect(component.find(".card-title").text()).toBe(movieObject.title);
    })

    it("Check movie render fine", () => {
        expect(component).toMatchSnapshot();
    })

})