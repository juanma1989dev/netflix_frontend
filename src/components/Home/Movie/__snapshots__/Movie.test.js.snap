// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Check movie render fine 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    id="auayuasgsydaguagduavdghasvdhgsa"
    poster="www.google.com"
    title="La pelicula de test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-md d-flex aling-items-strech mt-5"
>
        <div
                className="card Movie_card"
        >
                <img
                        alt="Card image cap"
                        className="card-img-top"
                        src="www.google.com"
                />
                <div
                        className="card-body"
                >
                        <h5
                                className="card-title"
                        >
                                La pelicula de test
                        </h5>
                        <Link
                                className="btn btn-primary"
                                replace={false}
                                to="/movie/auayuasgsydaguagduavdghasvdhgsa"
                        >
                                Ver
                        </Link>
                </div>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card Movie_card"
>
          <img
                    alt="Card image cap"
                    className="card-img-top"
                    src="www.google.com"
          />
          <div
                    className="card-body"
          >
                    <h5
                              className="card-title"
                    >
                              La pelicula de test
                    </h5>
                    <Link
                              className="btn btn-primary"
                              replace={false}
                              to="/movie/auayuasgsydaguagduavdghasvdhgsa"
                    >
                              Ver
                    </Link>
          </div>
</div>,
        "className": "col-md d-flex aling-items-strech mt-5",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Card image cap"
              className="card-img-top"
              src="www.google.com"
/>,
            <div
              className="card-body"
>
              <h5
                            className="card-title"
              >
                            La pelicula de test
              </h5>
              <Link
                            className="btn btn-primary"
                            replace={false}
                            to="/movie/auayuasgsydaguagduavdghasvdhgsa"
              >
                            Ver
              </Link>
</div>,
          ],
          "className": "card Movie_card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Card image cap",
              "className": "card-img-top",
              "src": "www.google.com",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5
                  className="card-title"
>
                  La pelicula de test
</h5>,
                <Link
                  className="btn btn-primary"
                  replace={false}
                  to="/movie/auayuasgsydaguagduavdghasvdhgsa"
>
                  Ver
</Link>,
              ],
              "className": "card-body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "La pelicula de test",
                  "className": "card-title",
                },
                "ref": null,
                "rendered": "La pelicula de test",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Ver",
                  "className": "btn btn-primary",
                  "replace": false,
                  "to": "/movie/auayuasgsydaguagduavdghasvdhgsa",
                },
                "ref": null,
                "rendered": "Ver",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-md d-flex aling-items-strech mt-5"
>
          <div
                    className="card Movie_card"
          >
                    <img
                              alt="Card image cap"
                              className="card-img-top"
                              src="www.google.com"
                    />
                    <div
                              className="card-body"
                    >
                              <h5
                                        className="card-title"
                              >
                                        La pelicula de test
                              </h5>
                              <Link
                                        className="btn btn-primary"
                                        replace={false}
                                        to="/movie/auayuasgsydaguagduavdghasvdhgsa"
                              >
                                        Ver
                              </Link>
                    </div>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card Movie_card"
>
            <img
                        alt="Card image cap"
                        className="card-img-top"
                        src="www.google.com"
            />
            <div
                        className="card-body"
            >
                        <h5
                                    className="card-title"
                        >
                                    La pelicula de test
                        </h5>
                        <Link
                                    className="btn btn-primary"
                                    replace={false}
                                    to="/movie/auayuasgsydaguagduavdghasvdhgsa"
                        >
                                    Ver
                        </Link>
            </div>
</div>,
          "className": "col-md d-flex aling-items-strech mt-5",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Card image cap"
                className="card-img-top"
                src="www.google.com"
/>,
              <div
                className="card-body"
>
                <h5
                                className="card-title"
                >
                                La pelicula de test
                </h5>
                <Link
                                className="btn btn-primary"
                                replace={false}
                                to="/movie/auayuasgsydaguagduavdghasvdhgsa"
                >
                                Ver
                </Link>
</div>,
            ],
            "className": "card Movie_card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Card image cap",
                "className": "card-img-top",
                "src": "www.google.com",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h5
                    className="card-title"
>
                    La pelicula de test
</h5>,
                  <Link
                    className="btn btn-primary"
                    replace={false}
                    to="/movie/auayuasgsydaguagduavdghasvdhgsa"
>
                    Ver
</Link>,
                ],
                "className": "card-body",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "La pelicula de test",
                    "className": "card-title",
                  },
                  "ref": null,
                  "rendered": "La pelicula de test",
                  "type": "h5",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Ver",
                    "className": "btn btn-primary",
                    "replace": false,
                    "to": "/movie/auayuasgsydaguagduavdghasvdhgsa",
                  },
                  "ref": null,
                  "rendered": "Ver",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
